test/fn/create-users.spec.js

```diff
@@ -1,32 +1,269 @@
+      .then(() => {
+        assert.deepEqual(api.requestLog(), [
+          { method: 'GET', url: '/api/v1/users-info?' + querystring.stringify(qs), body: {} },
+          { method: 'POST', url: '/api/v1/users', body: todd },
+        ]);
+      });
+  });
 
-      .then(() =>
+  it('should create user with existent place and response from user-info', () => {
+    const testDir = `data/create-users/existing-place`;
+    api.giveResponses({ body: { total_docs: 1000, warn: false } }, { body: {} });
+    const todd = {
+      username: 'todd',
+      password: 'Secret_1',
+      roles: ['district-admin'],
+      place: 'place_uuid_here',
+      contact: {
+        c_prop: 'c_val_a'
+      },
+      name: 'Alice Example',
+      phone: '+123456789',
+    };
+
+    const qs = {
+      facility_id: todd.place,
+      role: JSON.stringify(todd.roles),
+    };
+
+    return assertDbEmpty()
+      .then(() => /* when */ createUsers(testDir, api.couchUrl))
+      .then(() => {
         assert.deepEqual(api.requestLog(), [
-          { method: 'POST', url: '/api/v1/users', body: todd }
-        ])
-      );
+          { method: 'GET', url: '/api/v1/users-info?' + querystring.stringify(qs), body: {} },
+          { method: 'POST', url: '/api/v1/users', body: todd },
+        ]);
+      });
+  });
+
+  it('should create user with existent place and not implemented user-info endpoint', () => {
+    const testDir = `data/create-users/existing-place`;
+    api.giveResponses({ status: 404, body: { code: 404, error: 'not_found' } }, { body: {} });
+    const todd = {
+      username: 'todd',
+      password: 'Secret_1',
+      roles: ['district-admin'],
+      place: 'place_uuid_here',
+      contact: {
+        c_prop: 'c_val_a'
+      },
+      name: 'Alice Example',
+      phone: '+123456789',
+    };
+
+    const qs = {
+      facility_id: todd.place,
+      role: JSON.stringify(todd.roles),
+    };
+
+    return assertDbEmpty()
+      .then(() => /* when */ createUsers(testDir, api.couchUrl))
+      .then(() => {
+        assert.deepEqual(api.requestLog(), [
+          { method: 'GET', url: '/api/v1/users-info?' + querystring.stringify(qs), body: {} },
+          { method: 'POST', url: '/api/v1/users', body: todd },
+        ]);
+      });
+  });
+
+  it('should require user input before creating user with too many docs', () => {
+    const testDir = `data/create-users/existing-place`;
+    api.giveResponses({ body: { total_docs: 12000, warn: true, limit: 10000 } }, { body: {} });
+    const todd = {
+      username: 'todd',
+      password: 'Secret_1',
+      roles: ['district-admin'],
+      place: 'place_uuid_here',
+      contact: {
+        c_prop: 'c_val_a'
+      },
+      name: 'Alice Example',
+      phone: '+123456789',
+    };
+
+    readLine.keyInYN.returns(true);
+    const qs = {
+      facility_id: todd.place,
+      role: JSON.stringify(todd.roles),
+    };
+
+    return assertDbEmpty()
```

Comment:
Yea, fully agree. I try to keep away from async/await because I like it so much better than Promises and I fear I will not want write a single Promise again.